x-pack/plugins/fleet/public/components/agent_enrollment_flyout/steps.tsx

```diff
@@ -5,69 +5,190 @@
+
+import { agentPolicyRouteService } from '../../../common';
+
+import { sendGetK8sManifest } from '../../hooks/use_request/k8s';
 
 import { AdvancedAgentAuthenticationSettings } from './advanced_agent_authentication_settings';
 import { SelectCreateAgentPolicy } from './agent_policy_select_create';
 
-export const DownloadStep = (hasFleetServer: boolean) => {
+export const DownloadStep = (hasFleetServer: boolean, isK8s: string, enrollmentAPIKey: string) => {
   const kibanaVersion = useKibanaVersion();
+  const core = useStartServices();
+  const settings = useGetSettings();
   const kibanaVersionURLString = useMemo(
     () =>
       `${semverMajor(kibanaVersion)}-${semverMinor(kibanaVersion)}-${semverPatch(kibanaVersion)}`,
     [kibanaVersion]
   );
+
+  const [yaml, setYaml] = useState<any | undefined>();
+  const [fleetServer, setFleetServer] = useState<string | ''>();
+  useEffect(() => {
+    async function fetchK8sManifest() {
+      try {
+        if (isK8s !== 'IS_KUBERNETES') {
+          return;
+        }
+        const fleetServerHosts = settings.data?.item.fleet_server_hosts;
+        let host = '';
+        if (fleetServerHosts !== undefined && fleetServerHosts.length !== 0) {
+          setFleetServer(fleetServerHosts[0]);
+          host = fleetServerHosts[0];
+        }
+        const query = { fleetServer: host, enrolToken: enrollmentAPIKey };
+        const res = await sendGetK8sManifest(query);
+        if (res.error) {
+          throw res.error;
+        }
+
+        if (!res.data) {
+          throw new Error('No data while fetching agent manifest');
+        }
+
+        setYaml(res.data.item);
+      } catch (error) {
+        throw new Error('No data while fetching agent manifest');
+      }
+    }
+    fetchK8sManifest();
+  }, [isK8s, enrollmentAPIKey, settings.data?.item.fleet_server_hosts]);
+
+  const altTitle =
+    isK8s === 'IS_KUBERNETES'
+      ? i18n.translate('xpack.fleet.agentEnrollment.stepDownloadAgentForK8sTitle', {
+          defaultMessage: 'Download the Elastic Agent Manifest',
+        })
+      : i18n.translate('xpack.fleet.agentEnrollment.stepDownloadAgentTitle', {
+          defaultMessage: 'Download the Elastic Agent to your host',
+        });
   const title = hasFleetServer
     ? i18n.translate('xpack.fleet.agentEnrollment.stepDownloadAgentForFleetServerTitle', {
         defaultMessage: 'Download the Fleet Server to a centralized host',
       })
-    : i18n.translate('xpack.fleet.agentEnrollment.stepDownloadAgentTitle', {
-        defaultMessage: 'Download the Elastic Agent to your host',
-      });
+    : altTitle;
+
+  const altDownloadDescription =
+    isK8s === 'IS_KUBERNETES' ? (
+      <FormattedMessage
+        id="xpack.fleet.agentEnrollment.downloadDescriptionForK8s"
+        defaultMessage="Copy or download the Kubernetes manifest inside the Kubernetes cluster. Check {FleetUrlVariable} and {FleetTokenVariable} in the Daemonset environment variables and apply the manifest."
```

Comment:
I'm wondering if it could be misleading if user deployed Agent using ECK? I'm not sure what the "Kubernetes manifest" would be in that case ? _(not sure I have the full picture, so my comment might not be relevant)_.
CC @Kushmaro

Reply:
There is a point in what you are saying. ECK process is different and uses different manifests https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-eck.html.  But this could be an extra addition in the instructions. With this PR I am just implementing https://github.com/elastic/kibana/issues/92113

Reply:
IMHO we should promote both "regular" k8s manifests, however, we may want to promote ECK manifests  (or related messaging) prior.
What I mean here - is that when a customer would like to "run something elastic with k8s" we should first let them know that ECK is the best way of doing so. regardless of where such a message exists.

Reply:
@Kushmaro Can you clarify what you meant by - > "both "regular" k8s manifests"  and 
"run something elastic with k8s"  

If I understood @barkbay correctly, then the add->agent flyout should also generate agent configuration yaml and set of instructions to be followed in ECK user journey. The current solution assumes vanilla K8s setup, not associated with how the stack components are deployed. 

@barkbay @Kushmaro are you able to create another issue in this repo for accounting for ECK user requirement? 

CC @mlunadia